technical-reports/color/color-type.md

```diff
@@ -1,22 +1,768 @@
-# The color type
+# Color type
 
-Represents a 24bit RGB or 24+8bit RGBA color in the sRGB color space. The `$type` property MUST be set to the string `color`. The value MUST be a string containing a hex triplet/quartet including the preceding `#` character. To support other color spaces, such as HSL, translation tools SHOULD convert color tokens to the equivalent value as needed.
+Represents a color.
 
-For example, initially the color tokens MAY be defined as such:
+## Format
+
+For color tokens, the `$type` property MUST be set to the string `color`.
+
+The `$value` property can then be used to specify the details of the color, The `$value` object contains the following properties:
+
+- `colorSpace` (required): A string that specifices the color space. For supported color spaces, see the [supported color spaces](#supported-color-spaces) section below.
+- `components` (required): An array representing the color components. The number of components depends on the color space. Each element of the array MUST be either:
+  - A number
+  - The 'none' keyword
+- `alpha` (optional): A number that represents the alpha value of the color. This value is between `0` and `1`, where `0` is fully transparent and `1` is fully opaque.
+- `hex` (optional): A string that represents a fallback value of the color. The fallback color MUST be formatted in [6 digit CSS hex color notation](https://www.w3.org/TR/css-color-4/#hex-notation) format to avoid conflicts with the provided alpha value.
 
 <aside class="example">
 
 ```json
 {
-  "Majestic magenta": {
-    "$value": "#ff00ff",
-    "$type": "color"
+  "Hot pink": {
+    "$type": "color",
+    "$value": {
+      "colorSpace": "srgb",
+      "components": [1, 0, 1],
+      "alpha": 1,
+      "hex": "#ff00ff"
+    }
   },
   "Translucent shadow": {
-    "$value": "#00000080",
-    "$type": "color"
+    "$type": "color",
+    "$value": {
+      "colorSpace": "srgb",
+      "components": [0, 0, 0],
+      "alpha": 0.5,
+      "hex": "#000000"
+    }
+  }
+}
+```
+
+</aside>
+
+### The `none` keyword
+
+The `none` keyword MAY be used in the `components` array to indicate that a component is not applicable or not specified. This is useful for colors that do not require all components to be specified.
+For example, in the HSL color space, the `none` keyword MAY be used to indicate that there is no angle value for the color, which will be interpretted differently from a color with a hue angle of 0.
```

Comment:
We need to explain the concept of missing colors and reference the inspiration behind it: https://www.w3.org/TR/css-color-4/#missing